src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveInputStream.java

```diff
@@ -320,14 +445,32 @@ public ZipArchiveInputStream(final InputStream inputStream, final String encodin
      *                                             this to true if you want to read a split archive.
      * @since 1.20
      */
-    public ZipArchiveInputStream(final InputStream inputStream, final String encoding, final boolean useUnicodeExtraFields,
-            final boolean allowStoredEntriesWithDataDescriptor, final boolean skipSplitSig) {
-        super(inputStream, encoding);
+    public ZipArchiveInputStream(
+            final InputStream inputStream,
+            final String encoding,
+            final boolean useUnicodeExtraFields,
+            final boolean allowStoredEntriesWithDataDescriptor,
+            final boolean skipSplitSig) {
+        this(
+                inputStream,
+                builder()
+                        .setCharset(encoding)
+                        .setUseUnicodeExtraFields(useUnicodeExtraFields)
+                        .setAllowStoredEntriesWithDataDescriptor(allowStoredEntriesWithDataDescriptor)
+                        .setSkipSplitSig(skipSplitSig));
+    }
+
+    protected ZipArchiveInputStream(final AbstractBuilder<?, ?> builder) throws IOException {
```

Comment:
Javadoc.

Reply:
Fixed in https://github.com/apache/commons-compress/pull/706/commits/e6d2e40fd8b570d15c380c33f99f5e5ff9e72eb3